databases/week3/MealSharingApp_db.sql

```diff
@@ -0,0 +1,190 @@
+       ('Lunch','Pasta, vegetables and chicken drums','Københavan H','2021-08-12 13:30:00','4','1000.00'),
+       ('Snacks','Fried noodles with eggs','Københavan Ø','2021-09-10 16:00:00','3','450.00');
+
+-- get all meals
+SELECT * FROM meal;
+
+-- add new meal
+INSERT INTO meal (`title`,`description`,`location`,`when`,`max_reservations`,`price`)
+VALUES ('Evening drinks','Beer, wine, cola and soft drinks','Københavan V','2021-10-16 20:00:00','5','4500.00');
+
+-- get a meal with any id fx. 1 
+SELECT * FROM meal WHERE id = 3;
+
+-- update a meal with any id , update any attribute or multiple 
+UPDATE meal SET `location` = 'Århus', `description` = 'Pizza, salats and drinks'
+WHERE id = LAST_INSERT_ID();
+
+UPDATE meal SET `description` = 'soft and hard drinks'
+WHERE id = LAST_INSERT_ID();
+
+-- delete a meal with any Id
+DELETE FROM meal
+WHERE id = 2;
+
+-- Queries : reservation TABLE
+-- get all reservations
+SELECT * FROM reservation;
+
+-- add a new reservation
+INSERT INTO reservation (`no_of_guests`,`meal_id`,`phone_number`,`contact_name`,`contact_email_id`)
+VALUES ('12','1','123456','Michael','michael@mail.com');
+
+INSERT INTO reservation (`no_of_guests`,`meal_id`,`phone_number`,`contact_name`,`contact_email_id`)
+VALUES ('20','3','1235456','Michael & John','john@mail.com'),
+('8','4','3232444','Sara','sara@mail.com');
+
+-- get a reservarion with any ID
+SELECT * FROM reservation
+WHERE id = LAST_INSERT_ID();
+
+SELECT * FROM reservation
+WHERE id = 1;
+
+-- update a reservation with any id , update some attributes too
+UPDATE reservation
+SET `contact_name` = 'Bikram',
+`phone_number` = '111222',
+`contact_email_id` = 'bikram44@mail.com';
+
+ROLLBACK;
+
+UPDATE reservation
+SET `contact_name` = 'Alexander',
+`phone_number` = '456678',
+`contact_email_id` = 'alex44@mail.com'
+WHERE id = 6;
+
+UPDATE reservation
+SET `contact_name` = 'Marta',
+`phone_number` = '999222',
+`contact_email_id` = 'marta45@mail.com'
+WHERE id = 7;
+
+-- delete a reservation with any id 
+DELETE FROM reservation
+WHERE id = 6;
+
+UPDATE reservation
+SET `no_of_guests` = '5'
+WHERE id = 7;
+
+-- Queries: review TABLE
+-- add some reviews
+INSERT INTO review (`title`,`meal_id`,`stars`)
+VALUES ('Very Good','3','4'),
+('Excellent','4','5'),
+('Not good','1','2');
+ 
+ -- get all reviews
+ SELECT * FROM review
+ ORDER BY stars DESC;
+
+ -- get review, update and delete review
+SELECT title AS review_title, stars AS review_stars
+FROM review
+WHERE id = LAST_INSERT_ID();
+
+UPDATE review
+SET `title` = 'Good',
+`stars` = '3'
+WHERE id = 1;
+
+DELETE FROM review WHERE id = 2;
+
+-- Additional Queries
+-- get meals that has a price smaller than a specific price fx 90
+SELECT * FROM meal
+WHERE price < 1000;
+
+-- get meals that still has available reservations
+SELECT meal.title, meal.max_reservations,
+reservation.no_of_guests 
+FROM meal
+JOIN reservation ON meal.id = reservation.meal_id
+WHERE meal.max_reservations >= reservation.no_of_guests;
```

Comment:
Since a `Meal` can have multiple `Reservation`s, you should make a SUM of all the guests that are found for each reservation. 

Let me know if I should detail what I mean 😊

Reply:
you mean Select SUM(reservation.no_of_guests) From meal